### DISCUSSION TITLE
Add custom CSS file for component

### DISCUSSION THREAD

**Discussion by noahsb** [mantinedev/mantine#2549] [Q&A] [answered]
I have a unique situation where I need to import a css file for a single component that uses a 3rd party library that only provides styling via a `.css` import. 

Importing the css file does not look as expected because my provider is:

```
<MantineProvider withNormalizeCss withGlobalStyles>
```

Removing `withGlobalStyles` fixes the css issues, but I want to keep that for the rest of the files. 

Basically, I want a way to allow my css import to override all other css defaults for a single file because that's the only way to get the styling imported from a 3rd party library I'm using. 

Thanks!

**Comment by rtivital** [comment, marked as answer]
You need to make sure that CSS you are including is inserted after Mantine styles, see emotion cache options – https://mantine.dev/theming/emotion-cache/

If that does not work, you can try:
- copying styles of third-party library to your project, maybe your styles will have higher priority
- removing `withNormalizeCss` and importing it on your side as CSS file – this way you will be able to control the order of imports

**Reply by noahsb** [reply]
Thanks for the response! Both of the fallback options you gave result in the same effect. I do have most of the css styles working, but buttons with svg icons are clear. 

Can you go into a little more detail on how to insert the 3rd parties css after Mantine styles? I tried giving the emotion cache a different custom prefix and changing the value of `prepend` both without success

**Reply by rtivital** [reply]
Nope, cannot give any additional information, your issue is not related to Mantine – you need to figure out how to insert styles of third-party components after Mantine styles. From our side we provide custom emotion cache support which allows configuration of Mantine styles, look through documentation for your bundler/framework to understand how styles are being inserted into the HTML.

**Reply by noahsb** [reply]
Thanks for the info! I filed an issue in the library and it got fixed. For reference, they were having trouble with the box model reset that happens in `withGlobalStyles`

Relevant issue found here: 
https://github.com/graphql/graphiql/issues/2777
